backend/app-kubernetes-service/src/jvmMain/kotlin/services/proxy/TunnelManager.kt

```diff
@@ -55,36 +55,28 @@ class TunnelManager(private val k8: K8Dependencies) {
         cleanupExecutor.shutdownNow()
     }
 
-    suspend fun createOrUseExistingTunnel(id: String, remotePort: Int, rank: Int): Tunnel {
+    suspend fun createOrRecreateTunnel(id: String, remotePort: Int, rank: Int): Tunnel {
```

Comment:
Ideally this should only be recreated if the IP has changed. This code still works, because we never use the non kubernetes branch in production/development. But this does completely break the `!isRunningInsideKubernetes` branch which were present previously.

We should either completely get rid of the usage tracking and only use the direct route. Alternatively, we should change the code such that this is only re-created when the IP has changed.